src/display/text_layer.js

```diff
@@ -171,6 +171,7 @@ function appendText(task, geom, styles) {
     hasText: geom.str !== "",
     hasEOL: geom.hasEOL,
     fontSize: 0,
+    lang,
```

Comment:
If I look at #17770 it seems like the fix only aims to set the language of the canvas and not for the individual spans. Given that we usually have quite a number of spans in the document and they now all seem to contain this static value, is it necessary to have this here? In other words, if we just have to set the language once for the canvas in `getCtx`, I think we don't have to set it for the individual spans (and thereby saving some memory usage/cache size)?

Reply:
It just occurred to me that we *might* be able to "abuse" the `textDivProperties` WeakMap for this, which would reduce the number of copies to just one; I'll check if this works out.

Reply:
How about the updated patch, which removes the duplication by storing the `lang` attribute on the `container` DOM element?

Reply:
Yes, this seems better to me. Thanks!